database/pebble/db_test.go

```diff
@@ -0,0 +1,91 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package pebble
+
+import (
+	"testing"
+
+	"github.com/cockroachdb/pebble"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/utils/logging"
+)
+
+func TestInterface(t *testing.T) {
+	for _, test := range database.Tests {
+		folder := t.TempDir()
+		cfg := NewDefaultConfig()
+		db, err := New(folder, cfg, logging.NoLog{}, "pebble", prometheus.NewRegistry())
+		require.NoError(t, err)
+		defer db.Close()
+
+		test(t, db)
+
+		// The database may have been closed by the test, so we don't care if it
+		// errors here.
+		_ = db.Close()
+	}
+}
+
+func FuzzInterface(f *testing.F) {
+	for _, test := range database.FuzzTests {
+		folder := f.TempDir()
+		cfg := NewDefaultConfig()
+		db, err := New(folder, cfg, logging.NoLog{}, "", prometheus.NewRegistry())
+		require.NoError(f, err)
+		defer db.Close()
+
+		test(f, db)
+
+		// The database may have been closed by the test, so we don't care if it
+		// errors here.
+		_ = db.Close()
+	}
+}
+
+func BenchmarkInterface(b *testing.B) {
+	for _, size := range database.BenchmarkSizes {
+		keys, values := database.SetupBenchmark(b, size[0], size[1], size[2])
+		for _, bench := range database.Benchmarks {
+			folder := b.TempDir()
+			cfg := NewDefaultConfig()
+
+			db, err := New(folder, cfg, logging.NoLog{}, "", prometheus.NewRegistry())
+			require.NoError(b, err)
+			bench(b, db, "pebble", keys, values)
+
+			// The database may have been closed by the test, so we don't care if it
+			// errors here.
+			_ = db.Close()
+		}
+	}
+}
+
+func Test_bytesPrefix(t *testing.T) {
```

Comment:
I feel like instead of testing the `bytesPrefix` method (personally I am generally against testing non-exported code in a package) I think it would make more sense to implicitly test this by doing something like inserting a set of keys into the db and iterating over the range and verifying that the range we got from the iterator is what we expected